src/libflake/flake/flakeref.cc

```diff
@@ -234,15 +234,16 @@ std::optional<std::pair<FlakeRef, std::string>> parseURLFlakeRef(
         return std::nullopt;
     }
 
+    const auto subdir = getOr(parsedURL.query, "dir", "");
```

Comment:
What is missing here?

Reply:
If you're asking about the code: there should be nothing missing (unless _I_ am missing something). I left it as a draft since it seemed in some comments that there's no interest in a backport.